gateway/routes.go

```diff
@@ -0,0 +1,37 @@
+package gateway
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// Route is a specification and  handler for a REST endpoint.
+type Route struct {
+	verb string
+	path string
+	fn   func(http.ResponseWriter, *http.Request)
+}
+
+const APIVersion = "v1"
+
+func (g *gatewayAPI) Routes() []*Route {
+	return []*Route{
+		{verb: "POST", path: path("function", APIVersion), fn: g.uploadFunction},
+		{verb: "PUT", path: path("function/{id}", APIVersion), fn: g.updateFunction},
+		{verb: "GET", path: path("function/{id}", APIVersion), fn: g.getFunction},
```

Comment:
Add 
`{verb: "GET", path: path("function", APIVersion), fn: g.getFunctions},`

Should list all existing functions.